lib/grandpa/votes_tracker.go

```diff
@@ -138,6 +149,9 @@ func (vt *votesTracker) messages(blockHash common.Hash) (
 // as a slice of networkVoteMessages.
 func (vt *votesTracker) networkVoteMessages() (
 	messages []networkVoteMessage) {
+	vt.Lock()
+	defer vt.Unlock()
```

Comment:
nit this could be a rw mutex with read lock here

Reply:
https://github.com/ChainSafe/gossamer/pull/2923#discussion_r1021518744